core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -170,6 +170,10 @@ public void free() {
   public void reset() {
     if (consumer != null) {
       consumer.freeArray(array);
+      array = LongArray.empty;
```

Comment:
@hvanhovell ,
I'm starting to have second thoughts about the special `empty` instance here, I'm afraid the the nested call might trigger `freeArray` or something similar on it.
perhaps using null is a better option here?